test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js

```diff
@@ -0,0 +1,774 @@
+        // Case: Update for a Null attribute with string expression
+
+        var values = [
+            {
+                name: 'a',
+                type: 'NULL',
+                value: null
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/ws1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin5.json'))
+                .reply(204);
+        });
+
+        it('should apply the expression before sending the values', function(done) {
+            iotAgentLib.update('ws1', 'WeatherStationMultiple', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When an update comes for attributes without expressions and Boolean type', function() {
+        // Case: Update for a Boolean attribute without expression
+
+        var values = [
+            {
+                name: 'u',
+                type: 'Boolean',
+                value: true
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/light1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin9.json'))
+                .reply(204);
+        });
+
+        it('should apply the expression before sending the values', function(done) {
+            iotAgentLib.update('light1', 'Light', '', values, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When an update comes for attributes with numeric expressions and Boolean type', function() {
+        // Case: Update for a Boolean attribute with arithmetic expression
+
+        var values = [
+            {
+                name: 'u',
+                type: 'Boolean',
+                value: true
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities/ws1/attrs', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextRequests/updateContextExpressionPlugin10.json'))
```

Comment:
```
update: true -> ${@update * 20} -> ${ true * 20 } -> $ { true } -> true
```

However updateContextExpressionPlugin10.json shows false. Which step in my sequence is wrong?

Reply:
update: true -> ${@update * 20} -> ${ 1 * 20 } -> $ { 20 } -> $ { "20"} -> False 

Arithmetic expression so true is cast to integer ('1'). You are missing the cast to String at the end of the expression.

Reply:
If I'm understanding correctly, "20" (string) is casted to false (boolean) in JavaScript. However, in some point of the documentation it is said "everything with a "value" is true in JavasScript"... "20" is a "value", isn't it? Or maybe I'm missing something? :)

Reply:
Finally in order to do the final cast to boolean, we were using the following:

```
newAttribute.value = (newAttribute.value === 'true');
```

This implies that only the result of a String expression could finish in a true value and the only possible results that could be cast to true would be 'true'... not even '1'. So the documentation was not correct.

I have been thinking about how to override this problem. Finally, I have decided to use [boolean package](https://www.npmjs.com/package/boolean).

Reply:
Fixed in 65827ac053ce095effbb2d1fdf459eb596f38d27